src/scheduler/job_info.c

```diff
@@ -573,8 +609,18 @@ query_jobs(status *policy, int pbs_sd, queue_info *qinfo, resource_resv **pjobs,
 
 	server_time = qinfo->server->server_time;
 
+	for (i = 0; jobattrs[i] != NULL; i++) {
+		struct attrl *temp_attrl = NULL;
+
+		temp_attrl = new_attrl();
+		temp_attrl->name = strdup(jobattrs[i]);
+		temp_attrl->next = attrib;
+		temp_attrl->value = "";
+		attrib = temp_attrl;
```

Comment:
You are not freeing the members of the list. You may not have to do that if you make attrib a static link list.